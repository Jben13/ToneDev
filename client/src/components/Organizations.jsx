import React from 'react';
import {
  Link,
} from 'react-router-dom';
import axios from 'axios';
import Summary from './SummaryView';


class Organizations extends React.Component {
  constructor() {
    super();
    this.state = {
      orgs: [],
      showSummary: false,
    };
    this.renderView = this.renderView.bind(this);
  }

  // TODO: make a GET request to github microservice through nodeGateway display the org names
  componentDidMount() {
    axios.get('/api/gateway/github/orglist')
      .then((response) => {
        // console.log('ORGS GET: ', response.data);
        const orglist = response.data;
        const orgs = [];
        orglist.map((org) => {
          if (org.orgName !== null) {
            orgs.push(org.orgName);
          }
          return orgs;
        });
        this.setState({
          orgs,
        });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  // getOrgData(org) {

  // }

  renderView(name) {
    this.setState({
      showSummary: true,
      orgName: name.org,
    });
  }

  render() {
    const { orgs } = this.state;
    const { orgName } = this.state;
    const { showSummary } = this.state;

    if (showSummary === true) {
      return (
        <div>
          <div>
            <nav>
              <ul>
                <li><Link to="/">Home</Link></li>
                <li><Link to="/login">Login</Link></li>
              </ul>
            </nav>
          </div>
          <Summary orgName={orgName} />
        </div>
      );
    }
    return (
      <div>
        <div>
          <nav>
            <ul>
              <li><Link to="/">Home</Link></li>
              <li><Link to="/login">Logout</Link></li>
            </ul>
          </nav>
        </div>
        {orgs.map(org => (
          <div key={org}>
            <button type="submit" key={org} onClick={() => this.renderView({ org })}>
              {org}
            </button>
            <br />
          </div>
        ))}
      </div>
    );
  }
}


export default Organizations;
