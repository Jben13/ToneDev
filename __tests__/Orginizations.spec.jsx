import React from 'react';
import Organizations from '../client/src/components/Organizations';

test('App renders component correctly', () => {
  const component = global.shallow(<Organizations />);
  expect(component).toMatchSnapshot();
});
