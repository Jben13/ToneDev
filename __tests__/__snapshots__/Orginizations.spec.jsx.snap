// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/summary"
  >
    Google
  </Link>
  <br />
  <Link
    replace={false}
    to="/summary"
  >
    Twitter
  </Link>
  <br />
  <Link
    replace={false}
    to="/summary"
  >
    Facebook
  </Link>
  <br />
  <Link
    replace={false}
    to="/summary"
  >
    Microsoft
  </Link>
  <br />
  <Link
    replace={false}
    to="/summary"
  >
    Amazon
  </Link>
</div>
`;
